### PR TITLE
Introduce and apply new User Task `CORRECTED` event

### LINKED ISSUE #24756: Add a new `CORRECTED` event for `UserTask` records
We need to add a new intent that we can use to mark the event that a user task is corrected.

We'll also need to add a `UserTaskCorrectedApplier` to adjust the intermediate user task data with the provided information.

### PR DESCRIPTION
**Pull request by korthout**
## Description

Introduces a new `CORRECTED` intent and event applier. This stores the changes in the intermediate state, allowing consecutive task listeners to use this data or to roll them back by denying it.

Tests have been added to show that the event can be applied (i.e. appended) while the user task is: `CREATING`, `ASSIGNING`, `UPDATING`, `CANCELING`, or `COMPLETING`. Note that the test cases for `CREATING` and `UPDATING` are not supported yet and are skipped, but will be automatically supported when we introduce those listeners.

## Related issues

closes #24756

### CODE REVIEW

**Comment by korthout** [comment]
@ce-dmelnych @ana-vinogradova-camunda Please have a look if you have some time 🙇

File: zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/appliers/UserTaskCorrectedApplierTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+            .setDueDate("initial")
+            .setFollowUpDate("initial")
+            .setPriority(1);
+
+    setup.forEach(setupIntent -> testSetup.applyEventToState(userTaskKey, setupIntent, given));
+    testSetup.applyEventToState(userTaskKey, mapLifecycleStateToIntent(state), given);
+
+    Assumptions.assumeThat(userTaskState.getUserTask(userTaskKey)).isNotNull();
+    Assumptions.assumeThat(userTaskState.getLifecycleState(userTaskKey)).isEqualTo(state);
+    Assumptions.assumeThat(userTaskState.getIntermediateState(userTaskKey)).isNotNull();
```
Comment (ce-dmelnych) [inline zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/appliers/UserTaskCorrectedApplierTest.java:90]: Oh, I didn’t realise that `AssertJ` also provides the [Assumptions](https://assertj.github.io/doc/#assertj-core-assumptions) concept - good to know!
I've previously used [assumeThat](https://junit.org/junit4/javadoc/4.12/org/junit/Assume.html#assumeThat(T,%20org.hamcrest.Matcher)) from JUnit in some tests, but it’s not as convenient as the assumptions in `AssertJ`. TIL, thanks! 🙇

**Review by ce-dmelnych — APPROVED** [review]
LGTM! 👍 Great work!
Thanks for the tests that also cover the future implementation! 🙌

**Review by ana-vinogradova-camunda — APPROVED** [review]
LGTM! 🚀 
Great test coverage, thank you 👍

**Pull request merged**